devicetypes/smartthings/aeon-led-bulb-6.src/aeon-led-bulb-6.groovy

```diff
@@ -0,0 +1,337 @@
+		} else {
+			log.debug("Couldn't zwave.parse '$description'")
+		}
+	}
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	dimmerEvents(cmd)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicSet cmd) {
+	dimmerEvents(cmd)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelReport cmd) {
+	unschedule(offlinePing)
+	dimmerEvents(cmd)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchcolorv3.SwitchColorReport cmd) {
+	log.debug "got SwitchColorReport: $cmd"
+	state.colorReceived[cmd.colorComponent] = cmd.value
+	def result = []
+	def rgbNames = ["red", "green", "blue"]
+	def tempNames = ["warmWhite", "coldWhite"]
+	// Check if we got all the RGB color components
+	if (rgbNames.every { state.colorReceived[it] != null }) {
+		def colors = rgbNames.collect { state.colorReceived[it] }
+		log.debug "colors: $colors"
+		// Send the color as hex format
+		def hexColor = "#" + colors.collect { Integer.toHexString(it).padLeft(2, "0") }.join("")
+		result << createEvent(name: "color", value: hexColor)
+		// Send the color as hue and saturation
+		def hsv = rgbToHSV(*colors)
+		result << createEvent(name: "hue", value: hsv.hue)
+		result << createEvent(name: "saturation", value: hsv.saturation)
+		// Reset the values
+		rgbNames.collect { state.colorReceived[it] = null}
+	}
+	// Check if we got all the color temperature values
+	if (tempNames.every { state.colorReceived[it] != null}) {
+		def warmWhite = state.colorReceived["warmWhite"]
+		def coldWhite = state.colorReceived["coldWhite"]
+		log.debug "warmWhite: $warmWhite, coldWhite: $coldWhite"
+		if (warmWhite == 0 && coldWhite == 0) {
+			result = createEvent(name: "colorTemperature", value: COLOR_TEMP_MIN)
+		} else {
+			def parameterNumber = warmWhite ? WARM_WHITE_CONFIG : COLD_WHITE_CONFIG
+			result << response(command(zwave.configurationV2.configurationGet([parameterNumber: parameterNumber])))
+		}
+		// Reset the values
+		tempNames.collect { state.colorReceived[it] = null }
+	}
+	result
+}
+
+private dimmerEvents(physicalgraph.zwave.Command cmd) {
+	def value = (cmd.value ? "on" : "off")
+	def result = [createEvent(name: "switch", value: value, descriptionText: "$device.displayName was turned $value")]
+	if (cmd.value) {
+		result << createEvent(name: "level", value: cmd.value == 99 ? 100 : cmd.value , unit: "%")
+	}
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	if (encapsulatedCommand) {
+		def result = zwaveEvent(encapsulatedCommand)
+		result = result.collect {
+			if (it instanceof physicalgraph.device.HubAction && !it.toString().startsWith("9881")) {
+				response(cmd.CMD + "00" + it.toString())
+			} else {
+				it
+			}
+		}
+		result
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) {
+	log.debug "got ConfigurationReport: $cmd"
```

Comment:
you should verify that you're getting the report about the parameter you're expecting. 

there's no reason to expect it won't be, but better safe than sorry

Reply:
Done.